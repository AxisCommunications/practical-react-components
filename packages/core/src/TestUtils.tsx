import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { PracticalProvider } from './Practical'

/*
  Only used for Jest Snapshot Testing
*/

export const TP = ({
  children,
}: {
  readonly children: React.ReactElement
}): JSX.Element => <PracticalProvider>{children}</PracticalProvider>

/**
 * Mock reference to practical-root
 */
const createNodeMock = (element: React.ReactElement) => {
  if (element.props.id !== 'practical-root') {
    return null
  }

  return {}
}

export const TestRender = (jsx: JSX.Element) => {
  return renderer.create(<TP>{jsx}</TP>, { createNodeMock }).toJSON()
}

export const TestText = (): JSX.Element => <span>Test</span>

/* eslint-disable-next-line react/display-name */
export const TestTextWithForwardRef = React.forwardRef<HTMLElement>(
  ({ ...props }, ref) => (
    <span {...props} ref={ref}>
      Test
    </span>
  )
)
