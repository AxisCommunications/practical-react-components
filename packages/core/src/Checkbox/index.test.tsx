import React from 'react'
import 'jest-styled-components'

import { Checkbox } from '.'
import { TestRender } from '../TestUtils'

describe('Checkbox', () => {
  test('Checked', () => {
    const tree1 = TestRender(<Checkbox checked={false} />)
    expect(tree1).toMatchSnapshot()

    const tree2 = TestRender(<Checkbox checked={true} />)
    expect(tree2).toMatchSnapshot()

    const tree3 = TestRender(<Checkbox checked={true} partial={true} />)
    expect(tree3).toMatchSnapshot()
  })
})
