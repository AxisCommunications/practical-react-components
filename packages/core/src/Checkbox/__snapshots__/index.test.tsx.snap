// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Checked 1`] = `
.c4 {
  position: absolute;
  top: 3px;
  left: 3px;
  height: 18px;
  width: 18px;
  border-radius: 2px;
  -webkit-transition: -webkit-transform 100ms;
  -webkit-transition: transform 100ms;
  transition: transform 100ms;
  pointer-events: none;
}

.c2 {
  position: relative;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 24px;
  height: 24px;
}

.c2:hover .c3 {
  background-color: rgba(204,204,204,0.16);
  -webkit-transform: scale(1.77);
  -ms-transform: scale(1.77);
  transform: scale(1.77);
}

.c2:focus-within {
  outline: none;
}

.c2:focus-within .c3 {
  background-color: rgba(204,204,204,0.16);
  -webkit-transform: scale(1.77);
  -ms-transform: scale(1.77);
  transform: scale(1.77);
}

.c2:active .c3 {
  background-color: rgba(204,204,204,0.24);
  -webkit-transform: scale(1.88);
  -ms-transform: scale(1.88);
  transform: scale(1.88);
}

.c2:active .c5 {
  border: 2px solid rgb(61,61,61);
}

.c8 {
  position: absolute;
  top: 0;
  left: 0;
  cursor: pointer;
  display: unset;
  margin: 0;
  padding: 0;
  height: 100%;
  width: 100%;
  opacity: 0;
}

.c6 {
  width: 18px;
  height: 18px;
  border-radius: 2px;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  pointer-events: none;
  position: absolute;
}

.c7 {
  border: 2px solid rgb(102,102,102);
}

.c10 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c9 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-1"
    >
      <div
        checked={false}
        className="c2"
        disabled={false}
      >
        <div
          className="c3 c4"
        />
        <div
          className="c5 c6 c7"
        />
        <input
          checked={false}
          className="c8"
          disabled={false}
          onChange={[Function]}
          onFocus={[Function]}
          onPointerDown={[Function]}
          onPointerUp={[Function]}
          type="checkbox"
        />
      </div>
    </div>
  </div>
  <div>
    <div
      className="c9"
      id="layer-2"
    >
      <div
        className="c10"
      />
    </div>
  </div>
</div>
`;

exports[`Checkbox Checked 2`] = `
.c4 {
  position: absolute;
  top: 3px;
  left: 3px;
  height: 18px;
  width: 18px;
  border-radius: 2px;
  -webkit-transition: -webkit-transform 100ms;
  -webkit-transition: transform 100ms;
  transition: transform 100ms;
  pointer-events: none;
}

.c10:hover .c3 {
  background-color: rgba(204,204,204,0.16);
  -webkit-transform: scale(1.77);
  -ms-transform: scale(1.77);
  transform: scale(1.77);
}

.c10:focus-within .c3 {
  background-color: rgba(204,204,204,0.16);
  -webkit-transform: scale(1.77);
  -ms-transform: scale(1.77);
  transform: scale(1.77);
}

.c10:active .c3 {
  background-color: rgba(204,204,204,0.24);
  -webkit-transform: scale(1.88);
  -ms-transform: scale(1.88);
  transform: scale(1.88);
}

.c2 {
  position: relative;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 24px;
  height: 24px;
}

.c2:hover .c3 {
  background-color: rgba(103,58,183,0.16);
  -webkit-transform: scale(1.77);
  -ms-transform: scale(1.77);
  transform: scale(1.77);
}

.c2:focus-within {
  outline: none;
}

.c2:focus-within .c3 {
  background-color: rgba(103,58,183,0.16);
  -webkit-transform: scale(1.77);
  -ms-transform: scale(1.77);
  transform: scale(1.77);
}

.c2:active .c3 {
  background-color: rgba(103,58,183,0.24);
  -webkit-transform: scale(1.88);
  -ms-transform: scale(1.88);
  transform: scale(1.88);
}

.c7 {
  position: absolute;
  top: 0;
  left: 0;
  cursor: pointer;
  display: unset;
  margin: 0;
  padding: 0;
  height: 100%;
  width: 100%;
  opacity: 0;
}

.c5 {
  width: 18px;
  height: 18px;
  border-radius: 2px;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  pointer-events: none;
  position: absolute;
}

.c6 {
  background-color: rgb(103,58,183);
}

.c6::after {
  content: '';
  width: 5px;
  height: 10px;
  border: solid #fff;
  border-width: 0 2px 2px 0;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  margin-bottom: 3px;
}

.c9 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c8 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-3"
    >
      <div
        checked={true}
        className="c2"
        disabled={false}
      >
        <div
          className="c3 c4"
        />
        <div
          className="c5 c6"
        />
        <input
          checked={true}
          className="c7"
          disabled={false}
          onChange={[Function]}
          onFocus={[Function]}
          onPointerDown={[Function]}
          onPointerUp={[Function]}
          type="checkbox"
        />
      </div>
    </div>
  </div>
  <div>
    <div
      className="c8"
      id="layer-4"
    >
      <div
        className="c9"
      />
    </div>
  </div>
</div>
`;

exports[`Checkbox Checked 3`] = `
.c4 {
  position: absolute;
  top: 3px;
  left: 3px;
  height: 18px;
  width: 18px;
  border-radius: 2px;
  -webkit-transition: -webkit-transform 100ms;
  -webkit-transition: transform 100ms;
  transition: transform 100ms;
  pointer-events: none;
}

.c10:hover .c3 {
  background-color: rgba(204,204,204,0.16);
  -webkit-transform: scale(1.77);
  -ms-transform: scale(1.77);
  transform: scale(1.77);
}

.c10:focus-within .c3 {
  background-color: rgba(204,204,204,0.16);
  -webkit-transform: scale(1.77);
  -ms-transform: scale(1.77);
  transform: scale(1.77);
}

.c10:active .c3 {
  background-color: rgba(204,204,204,0.24);
  -webkit-transform: scale(1.88);
  -ms-transform: scale(1.88);
  transform: scale(1.88);
}

.c2 {
  position: relative;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 24px;
  height: 24px;
}

.c2:hover .c3 {
  background-color: rgba(103,58,183,0.16);
  -webkit-transform: scale(1.77);
  -ms-transform: scale(1.77);
  transform: scale(1.77);
}

.c2:focus-within {
  outline: none;
}

.c2:focus-within .c3 {
  background-color: rgba(103,58,183,0.16);
  -webkit-transform: scale(1.77);
  -ms-transform: scale(1.77);
  transform: scale(1.77);
}

.c2:active .c3 {
  background-color: rgba(103,58,183,0.24);
  -webkit-transform: scale(1.88);
  -ms-transform: scale(1.88);
  transform: scale(1.88);
}

.c7 {
  position: absolute;
  top: 0;
  left: 0;
  cursor: pointer;
  display: unset;
  margin: 0;
  padding: 0;
  height: 100%;
  width: 100%;
  opacity: 0;
}

.c5 {
  width: 18px;
  height: 18px;
  border-radius: 2px;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  pointer-events: none;
  position: absolute;
}

.c6 {
  background-color: rgb(103,58,183);
}

.c6::after {
  content: '';
  width: 10px;
  height: 2px;
  background-color: #fff;
  border-radius: 2px;
}

.c9 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c8 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-5"
    >
      <div
        checked={true}
        className="c2"
        disabled={false}
      >
        <div
          className="c3 c4"
        />
        <div
          className="c5 c6"
        />
        <input
          checked={true}
          className="c7"
          disabled={false}
          onChange={[Function]}
          onFocus={[Function]}
          onPointerDown={[Function]}
          onPointerUp={[Function]}
          type="checkbox"
        />
      </div>
    </div>
  </div>
  <div>
    <div
      className="c8"
      id="layer-6"
    >
      <div
        className="c9"
      />
    </div>
  </div>
</div>
`;
